### ISSUE TITLE
ROS2 release

### ISSUE THREAD

**Issue by wep21** [nobleo/paho.mqtt.cpp-release#1]
@Timple Do you have any plan to release this package on ros2(foxy/galactic/rolling)? I think your fork branch has already supported `ament_cmake`.

**Comment by Timple** [comment]
Can do!

I'll try to get this in before my hollidays. 

edit: Probably won't make that. I would first need to release `paho.mqtt.c` on those.

**Comment by wep21** [comment]
@Timple Thanks for PR!
When I tried to build paho.mqtt.cpp with `colcon`, the following build error occurred. Do you know why?
```
Starting >>> paho-mqtt-c
Finished <<< paho-mqtt-c [2.74s]                     
Starting >>> paho-mqtt-cpp
--- stderr: paho-mqtt-cpp                         
CMake Error at /usr/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:146 (message):
  Could NOT find PahoMqttC (missing: PAHO_MQTT_C_LIBRARIES)
Call Stack (most recent call first):
  /usr/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:393 (_FPHSA_FAILURE_MESSAGE)
  cmake/FindPahoMqttC.cmake:33 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  src/CMakeLists.txt:26 (find_package)


---
Failed   <<< paho-mqtt-cpp [0.42s, exited with code 1]

Summary: 1 package finished [3.27s]
  1 package failed: paho-mqtt-cpp
  1 package had stderr output: paho-mqtt-cpp
```
This is how to reproduce the error.
```
mkdir -p mqtt_ws/src
cd mqtt_ws/src
git clone https://github.com/nobleo/paho.mqtt.c.git -b catkin-support
git clone https://github.com/nobleo/paho.mqtt.cpp.git -b catkin-support
cd ../
source /opt/ros/galactic/setup.bash
colcon build --cmake-args -DCMAKE_BUILD_TYPE=Release --symlink-install
```

**Comment by Timple** [comment]
Compile with the cmake option `-DPAHO_WITH_SSL`?

**Comment by wep21** [comment]
This works! Thanks!
`colcon build --cmake-args -DCMAKE_BUILD_TYPE=Release -DPAHO_WITH_SSL=ON --symlink-install`

**Comment by Timple** [comment]
Tracking release here: https://github.com/ros/rosdistro/pull/31830

**Comment by Timple** [comment]
After the next galactic sync you should be able to do:

    sudo apt install ros-galactic-paho-mqtt-cpp

**Comment by vinnnyr** [comment]
@Timple  Is it possible to release into Rolling too? (not sure if this made sense to be a separate issue)

**Comment by Timple** [comment]
Not sure. Rolling targets a new ubuntu distro and paho.mqtt.c is already available there I think.

So that would need testing.

Did you test this package in ubuntu 22.04 already by compiling from source?
(Without compiling paho.mqtt.c from source?)

**Comment by vinnnyr** [comment]
The paho.mqtt.c does not seem to be in rosdep for `rolling` either: 

In  a fresh ros:rolling container:
`root@3dbb6fe57b06:/# rosdep db | grep mqtt

python-paho-mqtt-pip -> paho-mqtt
python3-paho-mqtt -> python3-paho-mqtt
`

**Comment by Timple** [comment]
I know. But that can be solved of course.

Can you find it in apt?
`apt-cache search paho`.

(Sorry, not on a linux machine now).

**Comment by Timple** [comment]
Seems like paho.mqtt.cpp can also be apt-installed: https://launchpad.net/ubuntu/+source/paho.mqtt.cpp

**Comment by vinnnyr** [comment]
You are right, seems to be apt installable:
```
root@3dbb6fe57b06:/# rosdep db | grep mqtt
python-paho-mqtt-pip -> paho-mqtt
python3-paho-mqtt -> python3-paho-mqtt
root@3dbb6fe57b06:/# apt-cache search paho
golang-github-eclipse-paho.mqtt.golang-dev - Go package implementing MQTT client library by Eclipse Paho
libpaho-mqtt-dev - Eclipse Paho MQTT C client - development files
libpaho-mqtt1.3 - Eclipse Paho MQTT C client - shared libraries
libpaho-mqttpp-dev - Eclipse Paho MQTT C++ Client Library - development files
libpaho-mqttpp3-1 - Eclipse Paho MQTT C++ Client Library - shared libraries
paho.mqtt.c-examples - Eclipse Paho MQTT C client - example files
pahole - set of advanced DWARF utilities
python3-asyncio-mqtt - Idomatic asyncio wrapper around paho-mqtt
python3-paho-mqtt - MQTT client class (Python 3)
```

So in that case, what is the purpose of this release repository?

**Comment by Timple** [comment]
The purpose of this release repository is to provide this package for all ROS versions on ubuntu <21.10.

For Humble/Rolling and beyond you can simply create a PR to https://github.com/ros/rosdistro to have it included as an apt target.

**Comment by vinnnyr** [comment]
ah ok did not realize this wasn't released in apt for under 21.10. Thanks!
